### PR TITLE
webgpu: support fft, ifft, rfft and irfft operators

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7033)

### CODE REVIEW

**Review by qjia7 — APPROVED** [review]
LGTM, thanks!

File: tfjs-backend-webgpu/src/fft_webgpu.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+      var result = 0.0;
+
+      for (var i = 0; i < uniforms.realShape[1]; i = i + 1) {
+        // x = (-2|2 * PI / N) * index * i;
+        let x = exponentMultiplierTimesIndexRatio * f32(i);
+        let expR = cos(x);
+        let expI = sin(x);
+        let real = getReal(batch, i);
+        let imag = getImag(batch, i);
```
Comment (hujiajie) [inline tfjs-backend-webgpu/src/fft_webgpu.ts:64]: Is this a direct mapping from the definition of discrete Fourier transform:
$$X_k = \sum_{n=0}^{N-1} x_n e^{-\frac{2kn\pi{}i}{N}}$$

It has $\mathcal{O}(N^2)$ complexity, so the first F in FFT ( $\mathcal{O}(N\log{}N)$ ) is lost. FFT should be super suitable for parallelism using the butterfly diagram, and could be a good example where WebGPU beats WebGL, though I have never remembered the details of the algorithm :sweat:

Comment (xhcao) [reply tfjs-backend-webgpu/src/fft_webgpu.ts:64]: Yes, It directly uses DFT to implement FFT here. FFT implementation is more difficult, I plan to implementation it in the future as an optimization task.

**Review by gyagp — APPROVED** [review]
LGTM

**Pull request merged**